action/lint.go

```diff
@@ -53,156 +49,112 @@ func LintAll(homedir string) {
+		return err == nil && stat.Mode().IsRegular()
+	})
 
-	if len(v.Warnings) > 0 || numErrors > 0 {
-		for _, warning := range v.Warnings {
-			log.Warn(warning)
-		}
-		for _, err := range v.Errors {
-			log.Err(err)
+	chartYamlValidation := chartYamlPresenceValidation.AddError("Chart.yaml is valid yaml", func(path string, v *validation.Validation) bool {
+		chartfile, err := v.Chartfile()
+		if err == nil {
+			cv.Chartfile = chartfile
 		}
 
-		if numErrors > 0 {
-			log.Err("Chart [%s] failed some checks", chartName)
-		} else {
-			log.Warn("Chart [%s] failed some checks", chartName)
-		}
-	} else {
-		log.Info("Chart [%s] has passed all necessary checks", chartName)
-	}
-}
+		return err == nil
+	})
 
-func directoryStructure(chartPath string, v *validation.Validation) map[string]os.FileInfo {
-	structure := make(map[string]os.FileInfo)
+	chartYamlNameValidation := chartYamlValidation.AddError("Chart.yaml has a name field", func(path string, v *validation.Validation) bool {
+		return cv.Chartfile.Name != ""
+	})
 
-	chartInfo, err := os.Stat(chartPath)
-	if err != nil {
-		v.AddError(fmt.Sprintf("Chart %s not found in workspace. Error: %v", chartPath, err))
-		return structure
-	}
+	chartYamlNameValidation.AddError("Name declared in Chart.yaml is the same as directory name.", func(path string, v *validation.Validation) bool {
+		return cv.Chartfile.Name == cv.ChartName()
+	})
 
-	if chartInfo.IsDir() {
-		files, _ := ioutil.ReadDir(chartPath)
-		for _, f := range files {
-			structure[f.Name()] = f
-		}
-	} else {
-		v.AddError(fmt.Sprintf("Chart Path [%s] is not a directory.", chartPath))
-	}
+	chartYamlValidation.AddError("Chart.yaml has a version field", func(path string, v *validation.Validation) bool {
+		return cv.Chartfile.Version != ""
+	})
 
-	return structure
-}
-
-func checkDirectoryStructure(structure map[string]os.FileInfo, chartPath string, v *validation.Validation) {
-	if _, ok := structure["README.md"]; ok != true {
-		v.AddWarning(fmt.Sprintf("A README file was not found in %s", chartPath))
-	}
+	chartYamlValidation.AddWarning("Chart.yaml has a description field", func(path string, v *validation.Validation) bool {
+		return cv.Chartfile.Description != ""
+	})
 
-	manifestInfo, ok := structure["manifests"]
-
-	if ok && manifestInfo.IsDir() {
-		// manifest files logic
-	} else {
-		v.AddError(fmt.Sprintf("A manifests directory was not found in %s", chartPath))
-	}
-}
-
-// verifyMetadata checks the Chart.yaml file for a Name, Version, Description, and Maintainers
-func verifyMetadata(chartPath string, v *validation.Validation) {
-	var y *chart.Chartfile
-
-	file := filepath.Join(chartPath, Chartfile)
-	chartDir := filepath.Base(chartPath)
-	b, err := ioutil.ReadFile(file)
-
-	if err != nil {
-		v.AddError("A Chart.yaml file was not found")
-		return
-	}
+	chartYamlValidation.AddWarning("Chart.yaml has a maintainers field", func(path string, v *validation.Validation) bool {
+		return cv.Chartfile.Maintainers != nil
```

Comment:
So it's ok for the `'Maintainers:` field to exist but be empty? (Not being snarky, just wasn't sure of the linting requirements.)